pkg/pipelines/yaml/resources_test.go

```diff
@@ -0,0 +1,56 @@
+package yaml
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/redhat-developer/kam/pkg/pipelines/namespaces"
+	res "github.com/redhat-developer/kam/pkg/pipelines/resources"
+	"github.com/spf13/afero"
+	"sigs.k8s.io/yaml"
+)
+
+func TestWriteResources(t *testing.T) {
+	fs := afero.NewOsFs()
+	homeEnv := "HOME"
+	originalHome := os.Getenv(homeEnv)
+	defer os.Setenv(homeEnv, originalHome)
+	path, cleanup := makeTempDir(t)
+	defer cleanup()
+	os.Setenv("HOME", path)
```

Comment:
This could use homeEnv ?

Reply:
we could also use a test for what happens if you don't provide `~` at the start of the filename?

Reply:
Updated. I've also added a case for writing to a path without permission